src/lib/x509/x509_ext.h

```diff
@@ -500,6 +500,65 @@ class OCSP_NoCheck final : public Certificate_Extension {
       void decode_inner(const std::vector<uint8_t>&) override;
 };
 
+/**
+* TNAuthList extension
+*
+* RFC8226 Secure Telephone Identity Credentials
+*   https://www.rfc-editor.org/rfc/rfc8226#section-9
+*/
+class BOTAN_PUBLIC_API(3, 5) TNAuthList final : public Certificate_Extension {
+   public:
+      class Entry final : public ASN1_Object {
+         public:
+            enum Type { TN_ServiceProviderCode = 0, TN_TelephoneNumberRange, TN_TelephoneNumber };
+
+            struct TelephoneNumberRange {
+                  ASN1_String start;  //TelephoneNumber (IA5String)
+                  size_t count;       //2..MAX
+            };
+
+            using RangeContainer = std::vector<TelephoneNumberRange>;
+            using DataContainer = std::variant<ASN1_String, RangeContainer>;
+
+            void encode_into(DER_Encoder&) const override;
+            void decode_from(class BER_Decoder& from) override;
+
+            Type type() const { return m_type; }
+
+            const std::string& service_provider_code() const { return std::get<ASN1_String>(m_data).value(); }
+
+            const RangeContainer& telephone_number_range() const { return std::get<RangeContainer>(m_data); }
+
+            const std::string& telephone_number() const { return std::get<ASN1_String>(m_data).value(); }
+
+            const DataContainer& data() { return m_data; }
```

Comment:
Is there a good reason to expose this to users? It seems like `type` plus the accessors gives the end user everything they need. It also seems to me error prone, vs the specific accessors which verify the type.

Reply:
removed data() accessor